cloudamqp/lavinmq
# Exchange abstraction

### WHAT is this pull request doing?
Move existing exchanges to AMQP namespace and introduce a clean base class for all exchanges, similar to what's been done with queues. This enables better separation of amqp and mqtt code.

### HOW can this pull request be tested?
Run specs to see that nothing breaks.


NOTE:
rebase+merge, not squash, to not screw up git history
